src/System.Runtime.Extensions/tests/System/IO/PathTestsBase.cs

```diff
@@ -201,6 +201,236 @@ public partial class PathTestsBase
             { @"C:\\foo2", @"C:\" },
         };
 
+        public static TheoryData<string, string, string> GetFullPath_Windows_NonFullyQualified()
+        {
+            string curDir = @"C:\git\corefx";
+            string root = @"C:\";
```

Comment:
This is really hard to follow. Why are you building the strings if you never change them?

Reply:
I am doing this to improve the readability of the tests and how these tests are related to Device path Tests. 
eg Tests containing relative segments in Qualified and Non_qualified just differ in the value of root and curDir.

Reply:
If there are correlations between sets of data that you want to show then you should pull them together. As it stands I can't clearly see any assertions you're trying to make in this regard. Show groupings by actually grouping into specific tests and/or test data sources.

Reply:
Another way to put why this is difficult is that I have to do the replacements in my head. Using comments to describe patterns that might not be obvious is easier to absorb.